var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -24,7 +25,9 @@ class Mvapich2(AutotoolsPackage):
     version('2.1', sha256='49f3225ad17d2f3b6b127236a0abdc979ca8a3efb8d47ab4b6cd4f5252d05d29')
 
     provides('mpi')
-    provides('mpi@:3.0')
+    provides('mpi@:3.1', when='@2.3:')
+    provides('mpi@:3.0', when='@2.2')
+    provides('mpi@:3.0', when='@2.1')
```

Comment:
Just put `provides('mpi@:3.0', when='@2.1:')` -- that covers 2.1 and 2.2.

Reply:
```
commit 1ad9aadecdab8dfc9fa879ff415562ccfeae1b5e
Author: Hari Subramoni <subramoni.1@osu.edu>
Date:   Thu Oct 17 11:59:07 2019 -0400

    Simplify rule so one rule covers 2.1 and 2.2

diff --git a/var/spack/repos/builtin/packages/mvapich2/package.py b/var/spack/repos/builtin/packages/mvapich2/package.py
index 326cdea..99ff779 100644
--- a/var/spack/repos/builtin/packages/mvapich2/package.py
+++ b/var/spack/repos/builtin/packages/mvapich2/package.py
@@ -26,7 +26,6 @@ class Mvapich2(AutotoolsPackage):

     provides('mpi')
     provides('mpi@:3.1', when='@2.3:')
-    provides('mpi@:3.0', when='@2.2')
     provides('mpi@:3.0', when='@2.1')

     variant('debug', default=False,
```